packages/firestore/src/platform/browser/webchannel_connection.ts

```diff
@@ -356,11 +359,28 @@ export class WebChannelConnection extends RestConnection {
       }
     );
 
+    let statEvent: StatEvent | null = null;
+    let hasRaisedTimeToFirstByteEvent = false;
+
     unguardedEventListen<StatEvent>(requestStats, Event.STAT_EVENT, event => {
       if (event.stat === Stat.PROXY) {
         logDebug(LOG_TAG, 'Detected buffering proxy');
+        statEvent = event;
       } else if (event.stat === Stat.NOPROXY) {
         logDebug(LOG_TAG, 'Detected no buffering proxy');
+        statEvent = event;
+      }
+
+      if (statEvent !== null && !hasRaisedTimeToFirstByteEvent) {
+        hasRaisedTimeToFirstByteEvent = true;
```

Comment:
Technically we can get the desired result exclusively with the if check in sync engine but then it looks weird b/c this component is frequently calling the onTimeToFirstByte and the elapsed time is going to keep getting bigger and bigger. It would get swallowed after the first invocation in sync engine but could make a future maintainer wonder what the intent was. Perhaps I should leave a comment instead?

Reply:
You could solve this by moving the call of `callOnTimeToFirstByte` to the if statements below. You will no longer need to check for `hasRaisedTimeToFirstByteEvent` and won't need to set it either. You can also remove `isLongPollingConnection:boolean `. Plus, if you initialize onOpenTimestamp to `new Date()` it will also not be nullable and you don't need to perform a debug check.

Reply:
I suggest something like this: https://gist.github.com/schmidt-sebastian/8faadec621a59b214b6cd01247de605a

Reply:
Got it. This makes sense and is more compact. Thank you.